src/main/java/com/google/devtools/build/lib/pkgcache/LoadingOptions.java

```diff
@@ -163,4 +175,72 @@
         + "then they can analyze test_suite targets."
   )
   public boolean expandTestSuites;
+
+  public static class BooleanFormulaConverter implements Converter<Expression> {
+    @Override
+    public Expression convert(String input) throws OptionsParsingException {
+      if (input.isEmpty()) {
+        return null;
+      }
+      if (isLegacySyntax(input)) {
+        input = convertLegacySyntaxToBooleanFormula(input);
+        GoogleLogger.forEnclosingClass().at(Level.WARNING).log(
```

Comment:
I do not see this logging in my environment. Is this the proper way to display warnings? If not what should I use instead?

Reply:
Just use System.err.println.